sdk/core/System.ClientModel/src/Convenience/ClientCache.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Linq;
+
+namespace System.ClientModel.Primitives;
+
+/// <summary>
+/// A cache for storing client instances, ensuring efficient reuse.
+/// Implements an LRU eviction policy.
+/// </summary>
+public class ClientCache
+{
+    private readonly ConcurrentDictionary<(Type, string), (object Client, long LastUsed)> _clients = new();
```

Comment:
I don't think we need a `ConcurrentDictionary` since all access to `_clients` is under lock.